index.js

```diff
@@ -11,8 +11,39 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @returns Promise
  */
 const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+
+  const filter = fetch(COLORS)
+    .then(response => response.json())
+    .then(colors => {
+      if (name != undefined) {
+        return colors.filter(
+          color => color.name.toLowerCase() === name.toLowerCase()
+        );
+      }
+      if (hex != undefined) {
+        return colors.filter(
+          color => color.hex.toLowerCase() === hex.toLowerCase()
+        );
+      }
+      if (compName != undefined) {
```

Comment:
If you use a single `filter` call, you can move all the filtering into that. This way if down the line we need to combine filters you can do so.